File: typescript/tests/wire/executionSessions.test.ts

import * as Sikaru from "../../src/api/index";
import { SikaruApi } from "../../src/Client";
import { mockServerPool } from "../mock-server/MockServerPool";

describe("ExecutionSessionsClient", () => {
    
    test("list (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.list("project_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("list (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.list("project_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("get (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions/session_id").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.get("project_id", "session_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("get (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions/session_id").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.get("project_id", "session_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("branch (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "idempotency_key" : "idempotency_key" , "source_run_id" : "source_run_id" };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-sessions/session_id/branches").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.branch("project_id", "session_id", {
    idempotency_key: "idempotency_key",
    source_run_id: "source_run_id"
});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("branch (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "idempotency_key" : "x" , "source_run_id" : "x" };
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-sessions/session_id/branches").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.branch("project_id", "session_id", {
    idempotency_key: "x",
    source_run_id: "x"
})
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("list_files (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions/session_id/files").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.listFiles("project_id", "session_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("list_files (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions/session_id/files").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.listFiles("project_id", "session_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("delete_file (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .delete("/v1/projects/project_id/execution-sessions/session_id/files/file_id").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.deleteFile("project_id", "session_id", "file_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("delete_file (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .delete("/v1/projects/project_id/execution-sessions/session_id/files/file_id").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.deleteFile("project_id", "session_id", "file_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("list_session_inputs (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions/session_id/inputs").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.listSessionInputs("project_id", "session_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("list_session_inputs (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-sessions/session_id/inputs").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.listSessionInputs("project_id", "session_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("append_turn (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "idempotency_key" : "idempotency_key" , "input" : { "key" : "value" } };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-sessions/session_id/turns").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.appendTurn("project_id", "session_id", {
    idempotency_key: "idempotency_key",
    input: {
        "key": "value"
    }
});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("append_turn (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "idempotency_key" : "x" , "input" : { "input" : { "key" : "value" } } };
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-sessions/session_id/turns").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.appendTurn("project_id", "session_id", {
    idempotency_key: "x",
    input: {
        "input": {
            "key": "value"
        }
    }
})
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("create (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "tenant_id" : "tenant_id" , "user_id" : "user_id" };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/execution-sessions").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionSessions.create("project_id", "harness_id", {
    tenant_id: "tenant_id",
    user_id: "user_id"
});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("create (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "tenant_id" : "x" , "user_id" : "x" };
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/execution-sessions").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionSessions.create("project_id", "harness_id", {
    tenant_id: "x",
    user_id: "x"
})
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
});

File: ruby/test/wire/execution_sessions_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ExecutionSessionsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_execution_sessions_list_with_wiremock
    test_id = "execution_sessions.list.0"

    @client.execution_sessions.list(
      project_id: "project_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.list.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-sessions",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_get_with_wiremock
    test_id = "execution_sessions.get.0"

    @client.execution_sessions.get(
      project_id: "project_id",
      session_id: "session_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.get.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-sessions/session_id",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_branch_with_wiremock
    test_id = "execution_sessions.branch.0"

    @client.execution_sessions.branch(
      project_id: "project_id",
      session_id: "session_id",
      idempotency_key: "idempotency_key",
      source_run_id: "source_run_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.branch.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/execution-sessions/session_id/branches",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_list_files_with_wiremock
    test_id = "execution_sessions.list_files.0"

    @client.execution_sessions.list_files(
      project_id: "project_id",
      session_id: "session_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.list_files.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-sessions/session_id/files",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_delete_file_with_wiremock
    test_id = "execution_sessions.delete_file.0"

    @client.execution_sessions.delete_file(
      project_id: "project_id",
      session_id: "session_id",
      file_id: "file_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.delete_file.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "DELETE",
      url_path: "/v1/projects/project_id/execution-sessions/session_id/files/file_id",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_download_file_with_wiremock
    test_id = "execution_sessions.download_file.0"

    @client.execution_sessions.download_file(
      project_id: "project_id",
      session_id: "session_id",
      file_id: "file_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.download_file.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-sessions/session_id/files/file_id/content",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_list_session_inputs_with_wiremock
    test_id = "execution_sessions.list_session_inputs.0"

    @client.execution_sessions.list_session_inputs(
      project_id: "project_id",
      session_id: "session_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.list_session_inputs.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-sessions/session_id/inputs",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_append_turn_with_wiremock
    test_id = "execution_sessions.append_turn.0"

    @client.execution_sessions.append_turn(
      project_id: "project_id",
      session_id: "session_id",
      idempotency_key: "idempotency_key",
      input: {
        key: "value"
      },
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.append_turn.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/execution-sessions/session_id/turns",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_sessions_create_with_wiremock
    test_id = "execution_sessions.create.0"

    @client.execution_sessions.create(
      project_id: "project_id",
      harness_id: "harness_id",
      tenant_id: "tenant_id",
      user_id: "user_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_sessions.create.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/harnesses/harness_id/execution-sessions",
      query_params: nil,
      expected: 1
    )
  end
end
